src/applications/letters/containers/VeteranBenefitSummaryOptions.jsx

```diff
@@ -0,0 +1,145 @@
+import {
+  benefitOptionsMap,
+  getFriendlyBenefitSummaryLabels,
+} from '../utils/helpers';
+import { UPDATE_BENEFIT_SUMMARY_REQUEST_OPTION } from '../utils/constants';
+
+const VeteranBenefitSummaryOptions = () => {
+  const dispatch = useDispatch();
+  const benefitsInfoSelector = useSelector(state => state.letters.benefitInfo);
+  const optionsLoadingSelector = useSelector(
+    state => state.letters.optionsLoading,
+  );
+  const requestOptionsSelector = useSelector(
+    state => state.letters.requestOptions,
+  );
+  const serviceInfoSelector = useSelector(state => state.letters.serviceInfo);
+
+  const handleChange = e => {
+    recordEvent({
+      // For Google Analytics
+      event: 'letter-benefit-option-clicked',
+      'letter-benefit-option': e.target.id,
+      'letter-benefit-option-status': e.target.checked
+        ? 'checked'
+        : 'unchecked',
+    });
+
+    dispatch({
+      type: UPDATE_BENEFIT_SUMMARY_REQUEST_OPTION,
+      propertyPath: benefitOptionsMap[e.target.id],
+      value: e.target.checked,
+    });
+  };
+
+  const renderBenefitsCheckboxes = () => {
+    const benefitKeys = Object.keys(benefitsInfoSelector);
+    const benefitCheckboxes = [];
+    const isVeteran = true;
+
+    benefitKeys.forEach(key => {
+      if (benefitsInfoSelector[key] === null) return;
+
+      const value = benefitsInfoSelector[key];
+      // Assume Lighthouse data from beneFitInformation is the source of truth
+      const displayOption = value !== false;
+      const labelText = getFriendlyBenefitSummaryLabels(key, isVeteran);
+
+      if (displayOption && labelText) {
+        benefitCheckboxes.push(
+          <li key={`option-${key}`} className="form-checkbox">
+            <input
+              checked={requestOptionsSelector[benefitOptionsMap[key]]}
+              id={key}
+              name={key}
+              type="checkbox"
+              onChange={e => handleChange(e)}
+            />
+            <label className="vads-u-margin-top--0" htmlFor={key}>
+              {labelText}
+            </label>
+          </li>,
+        );
+      }
+    });
+
+    return benefitCheckboxes;
+  };
+
+  const renderMilitaryServiceCheckbox = () => {
+    const militaryServiceRows = serviceInfoSelector || [];
+    const { militaryService } = requestOptionsSelector;
+
+    if (militaryServiceRows.length) {
+      return (
+        <li key="option-militaryService" className="form-checkbox">
+          <input
+            checked={militaryService}
+            id="militaryService"
+            name="militaryService"
+            type="checkbox"
+            onChange={e => handleChange(e)}
+          />
+          <label
+            name="militaryService-label"
+            className="vads-u-margin-top--0"
+            htmlFor="militaryService"
+          >
+            Military service information
+          </label>
+        </li>
+      );
+    }
+
+    return undefined;
+  };
+
+  switch (true) {
```

Comment:
This is a way to tackle a switch I have never seen before, I don't think 😆

Reply:
Initially I had 3 of 4 use cases mapped out for two Booleans, but last night I realized I needed to move some of that Toggler logic into the `DownloadLetterLink` component, so then there were two, hehe.

If this feels confusing or not a great way to handle the switch, I've got no issue updating it.